### PR TITLE
Add transaction details

### PR DESCRIPTION
**Pull request by SAnatoliiS**

### CODE REVIEW

File: kit/popups/src/Menu/UIMenuContainerContent.tsx [hunk 1]
```diff
@@ -45,6 +45,8 @@ function useMenuLocation(
 
         const boundaries = getBoundaries(windowDimensions, menuSize);
 
+        console.log({ boundaries });
```
Comment (savelichalex) [inline kit/popups/src/Menu/UIMenuContainerContent.tsx]: oops :)

File: stories/browser/src/types.ts [hunk 2]
```diff
@@ -299,11 +299,12 @@ export type TransactionConfirmationExternalState = {
...
     InteractiveMessageType.TransactionConfirmation,
     {
-        toAddress?: string;
-        onAddressPress?: () => void | Promise<void>;
-        recipientsCount?: number;
-        totalAmount: string | React.ReactElement<any, any>;
-        fees: string | React.ReactElement<any, any>;
+        recipient?: string | React.ReactElement<any, any>;
+        onRecipientPress?: () => void | Promise<void>;
+        amount: string | React.ReactElement<any, any>;
```
Comment (savelichalex) [inline stories/browser/src/types.ts]: I suggest to use `BigNumber` here instead of arbitrary React elements

File: stories/browser/src/types.ts [hunk 3]
```diff
@@ -299,11 +299,12 @@ export type TransactionConfirmationExternalState = {
...
+        action?: string | React.ReactElement<any, any>;
+        contractFee: string | React.ReactElement<any, any>;
+        networkFee: string | React.ReactElement<any, any>;
```
Comment (savelichalex) [inline stories/browser/src/types.ts]: Same, let's use `BigNumber`

File: stories/browser/src/types.ts [hunk 4]
```diff
@@ -299,11 +299,12 @@ export type TransactionConfirmationExternalState = {
 export type TransactionConfirmationMessage = InteractiveMessage<
     InteractiveMessageType.TransactionConfirmation,
     {
-        toAddress?: string;
-        onAddressPress?: () => void | Promise<void>;
-        recipientsCount?: number;
-        totalAmount: string | React.ReactElement<any, any>;
-        fees: string | React.ReactElement<any, any>;
+        recipient?: string | React.ReactElement<any, any>;
```
Comment (savelichalex) [inline stories/browser/src/types.ts]: not sure that it should accept React.Element too

File: stories/browser/src/types.ts [hunk 5]
```diff
@@ -299,11 +299,12 @@ export type TransactionConfirmationExternalState = {
...
+        onRecipientPress?: () => void | Promise<void>;
+        amount: string | React.ReactElement<any, any>;
+        action?: string | React.ReactElement<any, any>;
```
Comment (savelichalex) [inline stories/browser/src/types.ts]: what is action? 🤔

Comment (SAnatoliiS) [reply stories/browser/src/types.ts]: image

Comment (savelichalex) [reply stories/browser/src/types.ts]: Got it. Let's keep it text only

File: stories/browser/src/types.ts (hunk 2 above)
Comment (SAnatoliiS) [reply stories/browser/src/types.ts]: I've actually been thinking about it too.

**Comment by SAnatoliiS** [comment]
Снимок экрана 2022-04-19 в 16 39 43
Снимок экрана 2022-04-19 в 16 39 31

**Review by savelichalex — APPROVED** [review]

File: localization/src/languages/ko.json [hunk 6]
```diff
@@ -54,16 +54,19 @@
             "UseSecurityCard": "보안 카드 사용"
         },
         "TransactionConfirmation": {
+            "Title": "Confirm to proceed",
```
Comment (kkkhesha) [inline localization/src/languages/ko.json]:
```suggestion
            "Title": "계속하려면 확인",
```

File: localization/src/languages/ko.json [hunk 7]
```diff
@@ -54,16 +54,19 @@
...
+            "Recipient": "Recipient",
```
Comment (kkkhesha) [inline localization/src/languages/ko.json]:
```suggestion
            "Recipient": "받는 사람",
```

File: localization/src/languages/ko.json [hunk 8]
```diff
@@ -54,16 +54,19 @@
...
+            "Amount": "Amount",
```
Comment (kkkhesha) [inline localization/src/languages/ko.json]:
```suggestion
            "Amount": "액수",
```

File: localization/src/languages/ko.json [hunk 9]
```diff
@@ -54,16 +54,19 @@
...
+            "Action": "Action",
```
Comment (kkkhesha) [inline localization/src/languages/ko.json]:
```suggestion
            "Action": "활동",
```

File: localization/src/languages/ko.json [hunk 10]
```diff
@@ -54,16 +54,19 @@
...
+            "ContractFee": "Contract fee",
```
Comment (kkkhesha) [inline localization/src/languages/ko.json]:
```suggestion
            "ContractFee": "컨트랙트 수수료",
```

File: localization/src/languages/ko.json [hunk 11]
```diff
@@ -54,16 +54,19 @@
...
+            "More": "더...",
+            "NetworkFee": "Network fee",
```
Comment (kkkhesha) [inline localization/src/languages/ko.json]:
```suggestion
            "NetworkFee": "네트워크 수수료",
```

File: localization/src/languages/ko.json [hunk 12]
```diff
@@ -54,16 +54,19 @@
...
+            "UpTo": "up to",
```
Comment (kkkhesha) [inline localization/src/languages/ko.json]:
```suggestion
            "UpTo": "up to",
```
```suggestion
            "UpTo": "최대",
```

**Review by sertony — APPROVED** [review]

**Comment by sertony** [comment]
@SAnatoliiS will there be a PR in Surf with integration?

**Comment by savelichalex** [comment]
> @SAnatoliiS will there be a PR in Surf with integration?

I think we should ask @BigRoma to help, Roma please help to integrate it :)

**Review by sertony — APPROVED** [review]

**Review by sertony — APPROVED** [review]

File: Example/src/screens/Browser.tsx [hunk 13]
```diff
@@ -399,33 +391,18 @@ function BrowserAddMenu({
...
-                                </UILabel>
-                            </UILabel>
-                        ),
+                        action: 'ReturnDeposit',
+                        amount: new BigNumber(0.5),
+                        amountCurrency: 'SURF',
+                        contractFee: new BigNumber(5),
+                        contractFeeCurrency: 'EVER',
+                        networkFee: new BigNumber(0.12),
+                        networkFeeCurrency: 'EVER',
```
Comment (BigRoma) [inline Example/src/screens/Browser.tsx:405]: I think we can use interface for that props, like:
```typescript
interface TransactionItem { 
  value: BigNumber;
  currency: string;
  prefix?: string;
  folded?: boolean;
}
```
With helpful that variant we can shorter a code in main component.

File: stories/browser/src/TransactionDetails.tsx [hunk 14]
```diff
@@ -0,0 +1,204 @@
...
+            />
+            <NumberParameter
+                label={uiLocalized.Browser.TransactionConfirmation.ContractFee}
+                value={contractFee}
+                signChar={contractFeeCurrency}
+                prefix={`${uiLocalized.Browser.TransactionConfirmation.UpTo} `}
+            />
+        </>
+    );
+}
```
Comment (BigRoma) [inline stories/browser/src/TransactionDetails.tsx:136]: Are we really need split components?

File: Example/src/screens/Browser.tsx (hunk 13 above)
Comment (BigRoma) [reply Example/src/screens/Browser.tsx:405]: Transaction's bubble know too much information about transactions, IMHO.

**Review by BigRoma — CHANGES_REQUESTED** [review]

File: Example/src/screens/Browser.tsx (hunk 13 above)
Comment (BigRoma) [reply Example/src/screens/Browser.tsx:405]: I suggest simplify component.

Comment (SAnatoliiS) [reply Example/src/screens/Browser.tsx:405]: _With helpful that variant we can shorter a code in main component._
I think the code in main component is short enough and don't need to be shorter. And it very simple and readable.

Comment (SAnatoliiS) [reply Example/src/screens/Browser.tsx:405]: _Transaction's bubble know too much information about transactions, IMHO._
There is only information about transaction.

File: stories/browser/src/TransactionDetails.tsx (hunk 14 above)
Comment (SAnatoliiS) [reply stories/browser/src/TransactionDetails.tsx:136]: Yes. It brings us a simple and readable code. And protect from layout animations bugs.

**Review by BigRoma — APPROVED** [review]

**Pull request merged**